packages/ics721/src/ibc_packet_receive.rs

```diff
@@ -232,19 +226,11 @@ fn create_callback_msg(
             load_nft_contract_for_class_id(deps.storage, local_class_id.to_string())
                 .map_err(|_| ContractError::NoNftContractForClassId(local_class_id.to_string()))
         } else {
-            // If its a creation action, we can use the instantiate2 function to get the nft contract
-            // we don't care of the contract is instantiated yet or not, as later submessage will instantiate it if its not.
-            // The reason we use instantiate2 here is because we don't know if it was already instantiated or not.
-
-            let cw721_code_id = CW721_CODE_ID.load(deps.storage)?;
-            // for creating a predictable nft contract using, using instantiate2, we need: checksum, creator, and salt:
-            // - using class id as salt for instantiating nft contract guarantees a) predictable address and b) uniqueness
-            // for this salt must be of length 32 bytes, so we use sha256 to hash class id
-            let mut hasher = Sha256::new();
-            hasher.update(local_class_id.as_bytes());
-            let salt = hasher.finalize().to_vec();
-
-            get_instantiate2_address(deps, env.contract.address.as_str(), &salt, cw721_code_id)
+            // here we know NFT contract is always instantiated, so it is safe getting it from storage
```

Comment:
Do we? 
Because `CLASS_ID_AND_NFT_CONTRACT_INFO` storage is only updated on the `create_voucher_and_channel_messages`, which is executed AFTER `create_callback_msg`, meaning when we try to load from storage here, its still might be missing which should panic, on first transferred NFT.

We should check here if we have it in storage, get it, if we don't, get the init2 of it, which works because if we don't have it storage, it means we are creating it right now, which means we use the correct code id for it.